src/sklearn_evaluation/plot/plot.py

```diff
@@ -1,13 +1,124 @@
 import json
 from pathlib import Path
+import abc
 
 from ..report.serialize import figure2html
 
 
+# NOTE: deprecate
 class Plot:
     def _repr_html_(self):
         return figure2html(self.figure)
 
     def dump(self, path):
         data = self._get_data()
         Path(path).write_text(json.dumps(data), encoding="utf-8")
+
+
+class AbstractPlot(abc.ABC):
+    """An abstract class for all class-based plots"""
+
+    @abc.abstractmethod
+    def __init__(self, *, name=None):
+        """
+        The constructor must take all input data required to create the plot and assign
+        it as attributes. e.g., ``self.value = value``, no other processing should
+        happen here. The input data must be already aggregated. Since users typically
+        create plots from raw data (such as ``y_true`` and ``y_pred``), the will often
+        use ``Plot.from_raw_data`` instead of this constructor.
+
+        The only suggested argument is ``name=None``, which shoould be used to
+        identify the plot (e.g., in the title), and in composed plots.
+
+        All arguments beyond the input data must be keyword-only (add a *
+        argument between the input and the rest of the arguments).
+        """
+        pass
+
+    @abc.abstractmethod
+    def plot(self, ax=None):
+        """
+        All plotting related code must be here with one optional argument ``ax=None``.
+        Must assign, ``self.ax_``, and ``self.figure_`` attributes and return ``self``.
+        """
+        pass
+
+    @classmethod
+    @abc.abstractmethod
+    def from_raw_data(cls):
+        """Takes raw unaggregated data, compute statistics and initializes the object.
+        This is the method that users typically use. (e.g., they pass ``y_true``, and
+        ``y_pred`` here, we aggregate and call the constructor).
+
+        Apart from input data, this method must have the same argument as the
+        constructor.
+
+        All arguments beyond the input data must be keyword-only (add a *
```

Comment:
The `*` is not specified in the [example](https://github.com/ploomber/sklearn-evaluation/blob/refactor/src/sklearn_evaluation/plot/_example.py).
Also, maybe we should specify what aggregated vs unaggregated data means?

Reply:
good point, added an example of aggregated  vs unaggregated